### DISCUSSION TITLE
Add "Random Playlist (from library)" Playlist

### DISCUSSION THREAD

**Discussion by jordandrako** [music-assistant/support#2314] [Feature requests and ideas]
I'm looking to replace my direct Sonos integration automations with this. The automation I had before was to use the sonos favorites sensor, which gave the playlist IDs of all my favorited playlists, and would choose a random ID to play on the speakers around the house, like so:
```yaml
media_content_id: "{{ state_attr('sensor.sonos_favorites', 'items') | list | random }}"
```
My curated playlists mostly consist of lo-fi music that's nice to have in the background during the day. I don't know a way to randomly pick a playlist from the home assistant media integration, or how to list them in a template, so I'd like to request a new automatic "Random Playlist" playlist to go alongside the "Random Album" and "Random Artist" playlists.

**Comment by Rudd-O** [comment]
Related: https://github.com/orgs/music-assistant/discussions/2287

**Comment by OzGav** [comment]
Maybe you could do what this guy did https://github.com/orgs/music-assistant/discussions/1637#discussioncomment-9462085 but with playlists instead of artist?

**Reply by jordandrako** [reply]
I did try that, and it does allow me to randomize a playlist, but it'll pull any playlist from my sources, and not the favorited and curated playlists I have already. For example, using mass search for "lofi", there are 2 or 3 christmas lofi mixes in there and it's not quite ideal to have Christmas music going on for at least 11 months of the year. Thanks for the suggestion though, it may be a viable solution for some!

**Reply by OzGav** [reply]
You could rename them in some unique way so that the search only selects the ones you want.

**Reply by jordandrako** [reply]
I wish I could, but most of the playlists my family likes are maintained by Spotify or other community members.

**Comment by ministryofsillywalks** [comment]
I'm that other "random album guy"
I went through a few other options before finding my solution.
How many playlists are you talking about?
A solution could be to create a script that puts out a random number and then calls a specified playlist based on what number the result was. 
This was my first approach but with up to 40 albums to randomize it was a pain.
But if you only have like 10 playlists you want to randomize this might be an option?

**Reply by jordandrako** [reply]
I can grab the IDs of all the playlists (about 32 currently) and randomize from there, but the maintenance is a pain; adding more is manual work on my end, whereas adding a favorite playlist is something anyone in my family with access to the shared Spotify account can do. That's actually what I already do for specific holidays: I have a "Mood" drop down with Yule or Samhain and none for standard music, and for those "mood"s I have specific IDs in a list to randomize from, but those are just 5-10 playlists and don't update as often.

**Comment by jordandrako** [comment]
Alternative idea/solution: Add the ability to limit mass searches to favorited items. This would effectively solve the need as well.

**Reply by ministryofsillywalks** [reply]
That sounds like a great solution.
Would also not play random songs I don't actually have in my library when doing a broader search.

**Reply by OzGav** [reply]
Limiting mass.search to only return in library items or favourite items has been added to the backlog for version 2.1

**Reply by jordandrako** [reply]
Amazing, thank you for being so open and active, this project is amazing.

**Reply by OzGav** [reply]
Limiting mass search by library has been added to integration 2024.9.1 and later

**Reply by aaronklem95** [reply]
I have a similar use case of wanting to use MA to select one random **favorited** playlist each morning to play throughout the day. I currently only have 5 favorited playlist, but then have other unfavorited playlists that I would not want to be brought back as a result. Here's my YAML 

action: music_assistant.get_library
data:
  favorite: true
  config_entry_id: 01JVD3401K8T0ZENYM4DTHPB0X
  media_type: playlist
  offset: 0
  limit: 1
  order_by: random

The problem is, I am quite often getting this response...
items: []
limit: 1
offset: 0
order_by: random
media_type: playlist

Which goes away if I go through and favorite all my playlists. So it seems like the random playlist it has chosen is not one of y favorites, but then BECAUSE it's not a favorite it doesn't display anything in items. 

So I guess what I'm wondering is if there is a way to change the order of operations so that it only pulls favorites playlists in the first place. 

Sorry if this is the wrong way/place to ask this. I'm new to HA and MA and have never posted on forums before.

**Reply by OzGav** [reply]
That is a bug which has been fixed on the current beta

**Comment by lindsaymarkward** [comment]
For anyone else with a similar wish/issue to me, wanting a new different random album again quickly... according to @marcelveldt via Discord chat:
> It's refreshed at every sync interval or when you explicitly push the refresh button

**Reply by OzGav** [reply]
It’s in the docs https://music-assistant.io/usage/#playlists

**Reply by mptpro** [reply]
Is it possible in a HA script to force a refresh of the "500 Random Tracks" playlist?

Basically, I want to randomize the playlist every time I start it from HA. I know I can change the interval, but sometimes I start the playlist in, say, the living room. Then an hour later I'm working the garage and use HA to play it there, but since the playlist hasn't refreshed yet, it will start from the beginning and play the same songs again.

**Reply by OzGav** [reply]
https://github.com/orgs/music-assistant/discussions/2403#discussioncomment-12265282

**Reply by mptpro** [reply]
> https://github.com/orgs/music-assistant/discussions/2403#discussioncomment-12265282

thank you!

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
